scripts/billing/sql/daily_compute_usage_cud_sud_coverage.sql

```diff
@@ -0,0 +1,156 @@
+#standardSQL
+
+-- Query: CUD + SUD Coverage Query
+-- Last Updated: 2019-08-02
+--
+-- This query calculates the cost and usage amount (E.g. core*hrs) for Compute Engine usage.
+-- This can be used to answer questions like: how much of my usage is covered by CUD and SUD?
+
+WITH
+usage_data as (
+  SELECT
+    CAST(DATETIME(usage_start_time, "America/Los_Angeles") AS DATE) as usage_date,
+    *
+  -- *****************************************************************
+  -- *** INSERT YOUR BILLING BQ EXPORT TABLE NAME ON THE NEXT LINE ***
+  -- *****************************************************************
+  FROM `committed-use-discount-test.billing.gcp_billing_export_v1_016676_F7EDE9_45E9C2`
+  WHERE service.description = "Compute Engine"
+  AND CAST(DATETIME(usage_start_time, "America/Los_Angeles") AS DATE) >= "2019-04-01"
+),
+
+-- create temporary table prices, in order to calculate unit price per (date, sku, region) tuple.
+-- Export table only includes the credit $ amount in the credit.amount field. We can get the credit
+-- usage amount (e.g. core hours) by dividing credit.amount ($) by unit price for that sku.
+-- This assumes that the unit price for the usage is equal to the unit price for the associated
+-- CUD credit. This should be correct, except in rare cases where unit price for that sku changed
+-- during the day (i.e. a price drop, change in spending-based discount %)
+-- It is necessary to do this in a separate table and join back into the main data set vs.
+-- separately on each usage line because some line items have CUD credit but no associated
+-- usage. We would not otherwise be able to calculate a unit price for these line items.
+prices AS (
+  SELECT  
+    usage_date,
+    sku.id AS sku_id,
+    location.region AS region,
+    -- calculate unit price per sku for each day. Catch line items with 0 usage to avoid divide by zero.
+    -- using 1 assumes that there are no relevant (CUD related) skus with cost but 0 usage, 
+    -- which is correct for current billing data
+    IF(SUM(usage.amount) = 0, 0, SUM(cost) / SUM(usage.amount)) as unit_price
+  FROM usage_data, UNNEST(credits) as cred
+  WHERE TRUE
+    AND cred.name LIKE "Committed%"
+    OR cred.name LIKE "Sustained%"
+  GROUP BY 1,2,3
+  ORDER BY 1,2,3
+),
+
+-- Temporary table to pull out CUD + SUD credits cost and usage data.
+-- Talculate usage amount (e.g. core*hrs) data by dividing costs by unit price.
+credit_data AS (
+  SELECT
+    usage_date,
+    region,
+    sku_id,
+    sku_description,
+    machine_spec,
+    project_id,
+    SUM(CUD_covered_usage) AS CUD_covered_usage,
```

Comment:
Even though it's an abbreviation, i'd probably lower-case CUD & SUD in the column names just for consistency.

Reply:
Done